### PR TITLE
fix: Created matchAll function wrapper for minimatch so dir globs are properly ignored

### LINKED ISSUE #429: replace minimatch with minimatch-all
https://www.npmjs.com/package/minimatch-all

### PR DESCRIPTION
**Pull request by JohnstonCode**
fixes #429

### CODE REVIEW

File: src/util/globMatch.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  matchBase?: boolean;
+  nocomment?: boolean;
+  nonegate?: boolean;
+  flipNegate?: boolean;
+}
+
+export function matchAll(
+  path: string,
+  patterns: string[],
+  opts: IMinimatchOptions = {}
```
Comment (edgardmessias) [inline src/util/globMatch.ts]: replace `IMinimatchOptions` with `minimatch.IOptions`

Comment (JohnstonCode) [reply src/util/globMatch.ts]: Will do

**Comment by JohnstonCode** [comment]
Yes tests are needed

**Comment by JohnstonCode** [comment]
@edgardmessias builds are passing can you re-review the changes, please.

**Comment by edgardmessias** [comment]
Ok for me

**Comment by JohnstonCode** [comment]
Approve then, please.

**Pull request merged**